### ISSUE TITLE
`unusual_byte_groupings` doesn't work at a sub-module level

### ISSUE THREAD

**Issue by Lucretiel** [rust-lang/rust-clippy#6478]
I tried this code:

```rust
// Two 36-bit integers
Mask {
    #[allow(unusual_byte_groupings)]
    mask: 0b111111_111111_111111_111111_000000_001100,

    #[allow(unusual_byte_groupings)]
    setting: 0b000000_000000_000000_000000_010101_010001,
}
```

I expected to see this happen: Expected the code to pass clippy without warnings.

Instead, this happened: The `unusual_byte_groupings` lint still triggered on those literals. This doesn't happen when `#![allow(unusual_byte_groupings)]` is added to the top of the file, at the module level.

### Meta

- `cargo clippy -V`: clippy 0.0.212 (eb4fc71 2020-12-17)
- `rustc -Vv`:

```
rustc 1.50.0-nightly (eb4fc71dc 2020-12-17)
binary: rustc
commit-hash: eb4fc71dc9024f15a0c9cc44bcc10c861e9d585e
commit-date: 2020-12-17
host: x86_64-unknown-linux-gnu
release: 1.50.0-nightly
  ```

**Comment by ebroto** [comment]
Looking at the reference page for [struct expressions](https://doc.rust-lang.org/reference/expressions/struct-expr.html), I would say outer attributes are not allowed in that position.

It works either with an inner attribute just after the curly brace...
```rust
#![allow(dead_code)]

struct Mask {
    mask: i64,
    setting: i64,
}

fn mask() -> Mask {
    Mask {
        #![allow(clippy::unusual_byte_groupings)]
        mask: 0b111111_111111_111111_111111_000000_001100,
        setting: 0b000000_000000_000000_000000_010101_010001,
    }
}

fn main() {}
```
([playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=db6d412f3150cb42cf01cfab490c7da0))

...or with outer attributes directly on the literal expressions, but that requires an unstable feature
```rust
#![feature(stmt_expr_attributes)]
#![allow(dead_code)]

struct Mask {
    mask: i64,
    setting: i64,
}

fn mask() -> Mask {
    Mask {
        mask: #[allow(clippy::unusual_byte_groupings)]
        0b111111_111111_111111_111111_000000_001100,
        setting: #[allow(clippy::unusual_byte_groupings)]
        0b000000_000000_000000_000000_010101_010001,
    }
}

fn main() {}
```
([playground](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=5ffadf3b8e804c1a58387f22ea4b43ec))

**Comment by ebroto** [comment]
I'm gonna go ahead and close this as it seems to work as intended an there's not much Clippy can do here.

Thanks for taking the time to report it anyway!
